src/Components/TemporaryDrawer.js

```diff
@@ -0,0 +1,305 @@
+import { auth } from "../firebase";
+
+const linksData = ["", "quizzes", "quizzesAI", "onboarding", "guide"];
+
+export default function TemporaryDrawer({
+  logoutButton,
+  aiResponse,
+  clearAIResponse,
+  onDrawerOpen,
+  sendMessage,
+  handleAuthStateChanged,
+  handleLogout,
+  isLoggedIn,
+  loading,
+  historicalLandmarks,
+  natureParks,
+  politicalLandmarks,
+  setSelectedLandmarks,
+  renderMapComponent,
+}) {
+  const [state, setState] = React.useState({
+    top: false,
+    left: false,
+    bottom: false,
+    right: false,
+  });
+  const [userMessage, setUserMessage] = React.useState("");
+  const [landmarksChange, setLandmarksChange] = React.useState(null);
+
+  const toggleDrawer = (anchor, open) => (event) => {
+    // toggleDrawer is a higher order function that accepts the left and true/false arguments and returns an event object (e.g., can be a keydown or tab/shift)
+    setLandmarksChange(null);
+    if (
+      event.type === "keydown" &&
+      (event.key === "Tab" || event.key === "Shift")
+    ) {
+      return;
+    }
+
+    setState({ ...state, [anchor]: open });
+  };
+
+  React.useEffect(() => {
+    if (aiResponse) {
+      // set the state of left to be true so that the drawer can be opened
+      setState({ ...state, left: true });
+    }
+  }, [aiResponse]);
+
+  React.useEffect(() => {
+    if (landmarksChange === true) {
+      setState({ ...state, left: false });
+    }
+
+    if (landmarksChange === false) {
+      // set the state of left to be true so that the drawer can be opened
+      setState({ ...state, left: true });
+    }
+  }, [landmarksChange]);
+
+  React.useEffect(() => {
+    handleAuthStateChanged();
+  }, [handleAuthStateChanged]);
+
+  const list = (anchor) => (
+    <Box
+      sx={{ width: anchor === "top" || anchor === "bottom" ? "auto" : 800 }}
+      role="presentation"
+      // onClick={toggleDrawer(anchor, false)}
+      // onKeyDown={toggleDrawer(anchor, false)}
+    >
+      <List>
+        <List className="drawer-links">
+          <IconButton
+            onClick={toggleDrawer(anchor, false)}
+            sx={{ marginLeft: "300px" }}
+          >
+            <CloseIcon />
+          </IconButton>
+
+          {linksData.map((text, index) => (
+            <ListItemButton key={text}>
+              <ListItemIcon>
+                <Link to={`/${text}`}>
+                  {text === "" ? (
+                    <IconButton onClick={toggleDrawer(anchor, false)}>
+                      <HomeIcon sx={{ margin: "10px" }} />
+                      Home
+                    </IconButton>
+                  ) : text === "quizzesAI" ? (
+                    <IconButton>
+                      <QuizIcon sx={{ margin: "10px" }} /> Quiz
+                    </IconButton>
+                  ) : text === "guide" ? (
+                    <IconButton>
+                      <MenuBookIcon sx={{ margin: "10px" }} /> Guide
+                    </IconButton>
+                  ) : null}
```

Comment:
Again with the nested ternary operator - very hard to read.
Better to define an object that contains your HTML

```
const drawerItems = {
    ["quizzesAI"]: (<IconButton>
                      <QuizIcon sx={{ margin: "10px" }} /> Quiz
                    </IconButton>),
    ["guide"]: ....
}
```

then use it accordingly:

{text && (drawerItems[text])